paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnawareBucketCompactionTopologyBuilder.java

```diff
@@ -0,0 +1,175 @@
+    private static final String COMPACTION_COMMITTER_NAME = "Compaction Committer";
+
+    private transient StreamExecutionEnvironment env;
+    private final String tableIdentifier;
+    private final AppendOnlyFileStoreTable table;
+    @Nullable private List<Map<String, String>> specifiedPartitions = null;
+    private boolean isContinuous = false;
+
+    public UnawareBucketCompactionTopologyBuilder(
+            StreamExecutionEnvironment env,
+            String tableIdentifier,
+            AppendOnlyFileStoreTable table) {
+        this.env = env;
+        this.tableIdentifier = tableIdentifier;
+        this.table = table;
+    }
+
+    public void withContinuousMode(boolean isContinuous) {
+        this.isContinuous = isContinuous;
+    }
+
+    public void withPartitions(List<Map<String, String>> partitions) {
+        this.specifiedPartitions = partitions;
+    }
+
+    public void build() {
+
+        // build source from UnawareSourceFunction
+        DataStreamSource<AppendOnlyCompactionTask> source = buildSource();
+
+        // from source, construct the full flink job
+        sinkFromSource(source);
+    }
+
+    private DataStreamSource<AppendOnlyCompactionTask> buildSource() {
+        UnawareBucketSourceFunction source = new UnawareBucketSourceFunction(table);
+        long scanInterval = table.coreOptions().continuousDiscoveryInterval().toMillis();
+        source.withStreaming(isContinuous);
+        source.withScanInterval(scanInterval);
+        source.withFilter(getPartitionFilter());
+
+        final StreamSource<AppendOnlyCompactionTask, UnawareBucketSourceFunction> sourceOperator =
+                new StreamSource<>(source);
+        return new DataStreamSource<>(
+                env,
+                new CompactionTaskTypeInfo(),
+                sourceOperator,
+                false,
+                COMPACTION_COORDINATOR_NAME + " -> " + tableIdentifier,
+                isContinuous ? Boundedness.CONTINUOUS_UNBOUNDED : Boundedness.BOUNDED);
+    }
+
+    private void sinkFromSource(DataStreamSource<AppendOnlyCompactionTask> input) {
+        String commitUser = UUID.randomUUID().toString();
+        Options conf = Options.fromMap(table.options());
+        Integer compactionWorkerParallelism =
+                conf.get(FlinkConnectorOptions.COMPACTION_PARALLELISM);
+
+        StreamExecutionEnvironment env = input.getExecutionEnvironment();
+        CheckpointConfig checkpointConfig = env.getCheckpointConfig();
+
+        boolean streamingCheckpointEnabled =
+                isContinuous && checkpointConfig.isCheckpointingEnabled();
+        if (streamingCheckpointEnabled) {
+            FlinkSink.assertCheckpointConfiguration(env);
+        }
+
+        // compaction worker stage, to execute compaction tasks
+        CommittableTypeInfo committableTypeInfo = new CommittableTypeInfo();
+        SingleOutputStreamOperator<Committable> compacted =
+                input.transform(
```

Comment:
No `PartitionTransformation`? Default is forward?

Reply:
Seems we don't need specified partitioner. Any AppendOnlyCompactionTask could be invoke by any Compaction Worker

Reply:
We should set rebalance by hard code.